### ISSUE TITLE
OpenSSL client get stuck for duplicated Encrypted handshake message during DTLS handshake

### ISSUE THREAD

**Issue by zhangsenWang** [openssl/openssl#18887]
### Summary
During DTLS handshake, in case of duplicated `Encrypted handshake` message, the client get stuck. Tested OpenSSL versions include `3.0.4` `3.0.2` `1.1.1p` `1.1.1a` `1.1.0l` `1.1.0h` `1.0.2g`

### Example handshake flow log
```
S <- C ClientHello
S -> C HelloVerifyRequest
S <- C ClientHello
S -> C ServerHello
S -> C Certificate
S -> C ServerKeyExchange
S -> C ServerHelloDone
S <- C ClientKeyExchange
S <- C ChangeCipherSpec
S <- C Encrypted handshake
S -> C NewSessionTicket
S -> C ChangeCipherSpec
S -> C Encrypted handshake
S -> C Encrypted handshake   (duplicated)
```
### Steps to reproduce

The issue can be observed during an interoperability test with Mbed TLS :

- fetch and build Mbed TLS  from https://github.com/Mbed-TLS/mbedtls.git;
- from the Mbed TLS base directory, start Mbed TLS server and Mbed TLS proxy and OpenSSL client:

1. UDP proxy
`./programs/test/udp_proxy server_addr=127.0.0.1 server_port=19838 listen_addr=127.0.0.1 listen_port=28629 seed=2 drop=8 delay=8 duplicate=8`
2. Mbed TLS server
`./programs/ssl/ssl_server2 server_addr=127.0.0.1 server_port=19838 allow_sha1=1 dtls=1 debug_level=2 crt_file=tests/data_files/server7_int-ca.crt key_file=tests/data_files/server7.key hs_timeout=250-60000 mtu=512 force_version=dtls12 dgram_packing=0`
3. OpenSSL client
`echo 'GET / HTTP/1.0' | OPENSSL_DIR/openssl s_client -connect 127.0.0.1:28629 -dtls1_2`

Expected behavior: Successful handshake
Observed behavior: Handshake get stuck.

### Relevant issue
[#6902](https://github.com/openssl/openssl/issues/6902)
In that issue, bug occurs on OpenSSL server. But here, it's client.

**Comment by mattcaswell** [comment]
This is due to s_client using blocking sockets by default. When the duplicated record arrives it is (correctly) dropped and we immediately try to read the next record. However because we are using blocking sockets there might not be another record waiting and therefore we hang waiting for data to arrive.

Fix for this in #18978.

**Comment by nhorman** [comment]
marking as inactive, will be closed at the end of 3.4 dev barring further input
